open_rarity/resolver/testset_resolver.py

```diff
@@ -205,90 +165,82 @@ def resolve_collection_data(
     OpenRarity scorers'"""
     for token_with_rarity in tokens_with_rarity:
         token_id = token_with_rarity.token.token_identifier.token_id  # type: ignore
-        try:
-            token_with_rarity.rarities.extend(
-                [
-                    RarityData(
-                        provider=RankProvider.OR_ARITHMETIC,
-                        rank=scores.arithmetic_scores[token_id][0],
-                        score=scores.arithmetic_scores[token_id][1],
-                    ),
-                    RarityData(
-                        provider=RankProvider.OR_GEOMETRIC,
-                        rank=scores.geometric_scores[token_id][0],
-                        score=scores.geometric_scores[token_id][1],
-                    ),
-                    RarityData(
-                        provider=RankProvider.OR_HARMONIC,
-                        rank=scores.harmonic_scores[token_id][0],
-                        score=scores.harmonic_scores[token_id][1],
-                    ),
-                    RarityData(
-                        provider=RankProvider.OR_SUM,
-                        rank=scores.sum_scores[token_id][0],
-                        score=scores.sum_scores[token_id][1],
-                    ),
-                    RarityData(
-                        provider=RankProvider.OR_INFORMATION_CONTENT,
-                        rank=scores.information_content_scores[token_id][0],
-                        score=scores.information_content_scores[token_id][1],
-                    ),
-                ]
-            )
-        except Exception:
-            logger.exception(
-                f"Error occured with OR rank calc for token {token_with_rarity.token}"
-            )
+
+        token_with_rarity.rarities.extend(
+            [
+                RarityData(
+                    provider=RankProvider.OR_ARITHMETIC,
+                    rank=scores.arithmetic_scores[token_id][0],
+                    score=scores.arithmetic_scores[token_id][1],
+                ),
+                RarityData(
+                    provider=RankProvider.OR_GEOMETRIC,
+                    rank=scores.geometric_scores[token_id][0],
+                    score=scores.geometric_scores[token_id][1],
+                ),
+                RarityData(
+                    provider=RankProvider.OR_HARMONIC,
+                    rank=scores.harmonic_scores[token_id][0],
+                    score=scores.harmonic_scores[token_id][1],
+                ),
+                RarityData(
+                    provider=RankProvider.OR_SUM,
+                    rank=scores.sum_scores[token_id][0],
+                    score=scores.sum_scores[token_id][1],
+                ),
+                RarityData(
+                    provider=RankProvider.OR_INFORMATION_CONTENT,
+                    rank=scores.information_content_scores[token_id][0],
+                    score=scores.information_content_scores[token_id][1],
+                ),
+            ]
+        )
```

Comment:
Is this a function that alters a list in the scope above it? This is probably bad practice and less predictable. Better to copy, modify and return a new list. Python's memory model with mutables can be super weird.

Reply:
Unless you have a `inplace=True` argument or something.

Reply:
Yeah I agree, but it's what was there before and almost all the ranking providers modify in place so it would require a bunch of changes - I left it alone simply since it's not in core library but just in testset resolver which is effectively just a script to compare across various ranking providers.

Reply:
I think `add_` or `augment_` is probably the way to do that but I think with python being such an OOP language anything that alters in place should be a class method and not an external function.